src/output.js

```diff
@@ -0,0 +1,142 @@
+const { groupBy, find, get, findLast } = require('lodash')
+const { IncomingWebhook } = require('@slack/webhook')
+const github = require('@actions/github')
+const { readFileSync, readdirSync } = require('fs')
+const { join } = require('path')
+
+/** @type {string} */
+const homeDir = get(process.env, 'HOME', '')
+/** @type {string} */
+const githubRepo = get(process.env, 'GITHUB_REPOSITORY', '')
+/** @type {string} */
+const githubSHA = get(process.env, 'GITHUB_SHA', '')
+
+/**
+ * @param {{type: 'slack', status: number, slackWebhookUrl: string, githubToken: string, staticDistDir: string}} params
+ */
+async function run({ type, ...args }) {
+  try {
+    if (type === 'slack') {
+      await slackNotification(args)
+    } else {
+      console.log('Unknown output type: ', type)
+    }
+  } catch (e) {
+    throw e
+  }
+}
+
+/**
+ * @param {{status: number, slackWebhookUrl: string, githubToken: string, staticDistDir: string}} params
+ * @return {Promise<*>}
+ */
+async function slackNotification({ status, slackWebhookUrl, githubToken, staticDistDir = '.lighthouseci' }) {
+  const conclusion = status === 0 ? 'success' : 'failure'
+  const webhook = new IncomingWebhook(slackWebhookUrl)
+  const color = status === 0 ? 'good' : 'danger'
+  const staticDistDirPath = join(homeDir, staticDistDir)
+  const LHRNameFromPath = getLHRNameFromPath(staticDistDirPath)
+
+  const assertionResults = readFileSync(join(staticDistDirPath, 'assertion-results.json'))
+  const groupedResults = groupBy(JSON.parse(assertionResults.toString()), 'url')
+
+  const attachments = Object.values(groupedResults).map(groupedResult => {
+    const results = groupedResult.map(
+      /**
+       * @param {{auditId: string, auditProperty: string, auditTitle: string, expected: string, operator: string, actual: string, url: string}} res
+       * @return {{title: string, value: string}}
+       */
+      res => ({
+        title: `${res.auditId}.${res.auditProperty}`,
+        value: `${res.auditTitle} \n _Expected ${res.expected} ${
+          res.operator === '<=' ? ' less then' : ' greater than'
+        } actual ${res.actual}_`
+      })
+    )
+
+    const fields = results.slice(0, 2)
+    fields.length > 0 &&
+      fields.push({
+        title: '...',
+        value: ''
+      })
+
+    return {
+      text: `${groupedResult.length + 1} result(s) for ${groupedResult[0].url}`,
+      color,
+      fields
+    }
+  })
+
+  const gistName = `lhci-action-lhr-${githubRepo.split('/').join('-')}.json`
+  const octokit = new github.GitHub(githubToken)
+  const gists = await octokit.gists.list()
```

Comment:
This also seems like it could be a `createGist()` function? Pass it some data and an octokit client, get a url back?